src/core/executors/ExecutorStore.ts

```diff
@@ -19,8 +20,21 @@ export class ExecutorStore {
 
   // call processDeltaQueue on all executors immediately
   public forceDeltaQueueProcessingOnAllExecutors(): void {
+    let didSubscriptionExecutorProcessDeltaQueue = false;
+
     Object.values(this.store).forEach((executor) => {
+      if (
+        executor instanceof SubscriptionExecutor &&
```

Comment:
Does the `ExecutorFactory` already ensure we don't have duplicate executors? Or if it doesn't take care of it shouldn't we just omit adding it instead?

Reply:
Under this PR, the factory will return the reference to the single static subscription executor three times and be mapped to via the three `ModelName` keys that correspond to it: `pushSubscriptions, emailSubscriptions, and smsSubscriptions`

There's probably a cleaner way to do this without hard-coding a check here

Reply:
Adding on to what rgomezp said:

The [ExecutorStore](https://github.com/OneSignal/OneSignal-Website-SDK/blob/main/src/core/executors/ExecutorStore.ts) creates an executor for each value in [ModelName](https://github.com/OneSignal/OneSignal-Website-SDK/blob/main/src/core/models/SupportedModels.ts)
And there are 3 names for subscriptions:
1.  Identity
2. Properties
3. PushSubscriptions
4. EmailSubscriptions
5. SmsSubscriptions

This Pr should ensure that there are no duplicate executors from the `ExecutorFactory`. But we could omit adding it in the first place having 1 `ModelName` for the subscription executor instead. I don't think this would heavy lift but I'm not sure.

Reply:
The `ModelName` enum is widely used, not just for the executor factory.

You would have to change the way the `ExecutorStore` indexes the executors to only have 3 keys. But this adds a point of maintenance in the future should we add new supported models.

Is there an issue arising from the differences in behavior between the WebSDK & iOS? Or is this PR aimed at bringing the SDKs to parity to address an outstanding issue?

Reply:
Having one flush that is controlled by the operation repo would make it easier to implement a JWT feature where we would like to pause the operation repo when we receive an invalid token. And with changing to 1 flush, I didn't think there would be a need to have 3 separate subscription executors for push, sms, email.

Reply:
> But this adds a point of maintenance in the future should we add new supported models.

Could you elaborate on what maintenance would be needed?

> You would have to change the way the ExecutorStore indexes the executors to only have 3 keys. 

I thought this change would consist of changing the enum and the previous references to use 1 subscription enum. What other things would need to be done?

Reply:
> Could you elaborate on what maintenance would be needed?
The executor store store interface looks like

```ts
type ExecutorStoreInterface = {
  [key in ModelName]?: OSExecutor;
};
```

If you change it to say
```ts
type ExecutorStoreInterface = {
  identity: OSExecutor;
  properties: OSExecutor;
  subscriptions: OSExecutor;
};
```

or similar you would have to update in two places should you add new model types. Not a huge deal though.

> I thought this change would consist of changing the enum and the previous references to use 1 subscription enum. What other things would need to be done?

The enum isn't just used for creating executors. It's used throughout the user-model SDK.

Reply:
I believe it would instead be changing this code:
```ts
export enum ModelName {
  Identity = 'identity',
  Properties = 'properties',
  // TO DO: make singular
  PushSubscriptions = 'pushSubscriptions',
  EmailSubscriptions = 'emailSubscriptions',
  SmsSubscriptions = 'smsSubscriptions',
}
```
to this:
```ts
export enum ModelName {
  Identity = 'identity',
  Properties = 'properties',
  Subscriptions = 'subscriptions',
}
```

I think this will simply things and means if we add another subscription type we won't have to change this again. We will have to update code that depends on `ModelName` however, and migrate any existing cached operations to the new name though.

Reply:
> I think this will simply things and means if we add another subscription type we won't have to change this again. We will have to update code that depends on ModelName however, and migrate any existing cached operations to the new name though.

That's a big lift. We also rely on the model name to differentiate between subscription models, like in the `CoreModuleDirector`, `OSModel`, etc...